pkgs/by-name/xd/xdg-desktop-portal-cosmic/package.nix

```
@@ -0,0 +1,55 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, mesa
+, libglvnd
+, libxkbcommon
+, pipewire
+, wayland
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "xdg-desktop-portal-cosmic";
+  version = "unstable-2023-10-11";
+
+  src = fetchFromGitHub {
+    owner = "pop-os";
+    repo = pname;
+    rev = "b0783c726a593ca44959c50d7306ac3581826dcb";
+    hash = "sha256-5rkbe/niEwO5XMS4hHJ9cElBM0X+nB5RYiUqmy7gR9s=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "smithay-client-toolkit-0.17.0" = "sha256-4CIssnGExC0w00voCrq7W6EJp2o0ndqgMvWuwBZITtw=";
+      "cosmic-protocols-0.1.0" = "sha256-MB/xdIYwvOTzvZFIKcPmccZFAgSdOvLmnndl10wj9BE=";
+      "ashpd-0.6.0" = "sha256-g1JvpdCTKpUjQHz+Hn0Zp8Ou5gJdBqYwQlkBo0xgXAU=";
+      "libspa-0.7.2" = "sha256-n8MYJuWMH1fcWS91eI2ErdEYMcda6D8Cxo5egKX/qB4=";
+    };
+  };
```

I would prefer to avoid  vendoring the 2500+ line cargo lock file, and instead just provide a `cargoHash`
```suggestion
  cargoHash = "xxx";
```

---

@jonringer This is not possible when the Cargo.lock contains git dependencies.

---

Yes, unfortunately we cannot avoid vendoring the `Cargo.lock` file here as our FOD fetcher for Cargo does not support `git` deps and recommends vendoring the file and using `cargoLock.lockFile`: https://github.com/NixOS/nixpkgs/blob/b31d979c26854fa62cc59b5b5649012666169e5c/pkgs/build-support/rust/fetch-cargo-tarball/default.nix#L79-L88